src/collections/Table/Table.js

```diff
@@ -10,43 +10,97 @@ export default class Table extends Component {
     children: ofComponentTypes(['TableColumn']),
     className: PropTypes.string,
     data: PropTypes.array,
+    onSelectRow: PropTypes.func,
     onSortChange: PropTypes.func,
+    selectedRows: PropTypes.arrayOf(PropTypes.number),
     sort: PropTypes.shape({
       key: PropTypes.string,
-      direction: PropTypes.string, // descending|ascending, defaults to descending
+      direction: PropTypes.oneOf(['descending', 'ascending']),
     }),
   };
 
+  static defaultProps = {
+    selectedRows: [],
+    sort: {
+      key: null,
+      direction: 'descending',
+    },
+  }
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      selectedRows: this.props.selectedRows,
+    };
+  }
+
   static getSafeCellContents(content) {
     // React cannot render objects, stringify them instead
     return _.isObject(content) ? JSON.stringify(content) : content;
   }
 
+  _isRowSelected(index) {
+    return _.includes(this.state.selectedRows, index);
+  }
+
+  _isSelectable = () => {
+    return _.includes(this.props.className, 'selectable');
+  };
+
+  _unselectRow(index) {
+    if (!this._isSelectable()) return;
+    this.setState({
+      selectedRows: _.without(this.state.selectedRows, index)
+    });
+  }
+
+  _selectRow(index) {
+    if (!this._isSelectable()) return;
+    this.setState({selectedRows: [index]});
+  }
+
+  _unselectAllRows() {
+    this.setState({selectedRows: []});
+  }
+
+  _toggleSelectRow(index) {
+    if (this._isRowSelected(index)) {
+      this._unselectRow(index);
+    } else {
+      this._selectRow(index);
+    }
+  }
+
+  _handleSelectRow(rowItem, rowIndex) {
+    this._toggleSelectRow(rowIndex);
+    if (this.props.onSelectRow) this.props.onSelectRow(rowItem, rowIndex);
+  }
+
   _handleSortHeaderChange(key, direction) {
     const {onSortChange} = this.props;
     if (onSortChange) {
+      this._unselectAllRows();
```

Comment:
Since sorting happens outside the component we can't track which items were selected reliably.  Instead, we simply clear selection on sort.